validation/schemas.ts

```diff
@@ -22,39 +22,77 @@ export const envSchema = z.object({
 export const queryParamsIdSchema = z.object({ id: z.uuid() });
 
 // Users
-export const userInsertSchema = createInsertSchema(userTable);
-export const userUpdateSchema = createUpdateSchema(userTable);
+export const userInsertSchema = createInsertSchema(userTable)
+  .extend({
+    email: z.email().nonempty(),
+  })
+  .omit({ id: true, invitedByUserId: true, createdAt: true, updatedAt: true });
```

Comment:
Why remove the invitedByUserId? It seems like sometimes this will be necessary when inserting.

Reply:
That makes sense, maybe you should rename this schema to `createUserRequestSchema` or something more specific. Eventually you may want to have other schemas that are based on the insert schema, so maybe both should be declared as variables:

```
export const userInsertSchema = createInsertSchema(userTable).extend({
  email: z.email().nonempty(),
})
export const createUserRequestSchema = userInsertSchema.omit({
  id: true, invitedByUserId: true, createdAt: true, updatedAt: true
});
// Full user insert schemas is available if more requests based on it are needed.
```

Reply:
Looks good.